### ISSUE TITLE
xsd in folder hierarchy results in XMLSchemaValueError

### ISSUE THREAD

**Issue by Skorpeo** [brunato/xmlschema#54]
When working with xsd in certain folder hierarchy leads to XMLSchemaValueError.  Files are attached in zip file with the folder hierarchy.

```
import xmlschema
path = "folder1/folder2b/folder2/folder3/usage.xsd"
xs = xmlschema.XMLSchema(path)
```
results in: `XMLSchemaValueError: a '{http://www.w3.org/2001/XMLSchema}list' definition required for XsdList(name='someSimpleType').`

[xsdtests.zip](https://github.com/brunato/xmlschema/files/1845087/xsdtests.zip)

**Comment by brunato** [comment]
This issue was related to a bug in general schema building with global "xs:list" simpleType declarations (and also for "xs:union"). Now, when the type's elem attribute is set, the class code searches for a xs:list inside the xs:simpleType declaration, so the error that you reported doesn't happen again.

The fix for this issue is available with v0.9.24.

Thanks

**Comment by Skorpeo** [comment]
thanks very much that worked!
